src/megatron/bridge/models/qwen_vl/modelling_qwen3_vl/utils.py

```
@@ -26,181 +184,572 @@ def split_deepstack_embs(
-                        video_grid_thw[video_index][0],
-                        video_grid_thw[video_index][1],
-                        video_grid_thw[video_index][2],
-                    )
-                    video_index += 1
-                    remain_videos -= 1
-                    ed = ed_video
-                llm_grid_t, llm_grid_h, llm_grid_w = (
-                    t.item(),
-                    h.item() // spatial_merge_size,
-                    w.item() // spatial_merge_size,
+            assert False, f"should not have {token_id=}"
```

Ruff (0.14.14)

[warning] Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)